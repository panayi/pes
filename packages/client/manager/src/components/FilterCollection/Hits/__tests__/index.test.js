import React from 'react';
import { PropTypes } from 'prop-types';
import { BaseHits } from '../index';

describe('FilterCollection', () => {
  describe('BaseHits', () => {
    const defaultProps = {
      hits: [],
      hitComponent: noop,
      itemHeight: 1,
      query: '',
    };

    it('should render hits', () => {
      const hits = [{ name: 'foo' }, { name: 'bar' }, { name: 'baz' }];
      const HitComponent = ({ hit }) => <span>{hit.name}</span>;
      HitComponent.propTypes = {
        hit: PropTypes.shape({}).isRequired,
      };
      const wrapper = mount(
        <BaseHits
          {...defaultProps}
          hits={hits}
          hitComponent={HitComponent}
          itemHeight={20}
          width={100}
          height={100}
        />,
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render <Nothing /> when hits is empty and query is empty', () => {
      const wrapper = mount(
        <BaseHits {...defaultProps} width={100} height={100} />,
      );
      expect(wrapper.type()).toBe(null);
    });

    it('should render noHits when hits is empty and query is non empty', () => {
      const noResults = () => <span>No Results Found</span>;
      const wrapper = mount(
        <BaseHits
          {...defaultProps}
          noHitsComponent={noResults}
          query="abc"
          width={100}
          height={100}
        />,
      );
      expect(
        wrapper.containsMatchingElement(<span>No Results Found</span>),
      ).toBe(true);
    });
  });
});
