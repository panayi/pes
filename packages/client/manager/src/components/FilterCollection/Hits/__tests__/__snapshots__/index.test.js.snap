// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterCollection BaseHits should render hits 1`] = `
<Component
  height={100}
  hitComponent={[Function]}
  hitProps={Object {}}
  hits={
    Array [
      Object {
        "name": "foo",
      },
      Object {
        "name": "bar",
      },
      Object {
        "name": "baz",
      },
    ]
  }
  itemHeight={20}
  noHitsComponent={null}
  onFirstHitChange={[Function]}
  query=""
  width={100}
>
  <List
    autoHeight={false}
    estimatedRowSize={30}
    height={100}
    noRowsRenderer={[Function]}
    onRowsRendered={[Function]}
    onScroll={[Function]}
    overscanIndicesGetter={[Function]}
    overscanRowCount={10}
    rowCount={3}
    rowHeight={20}
    rowRenderer={[Function]}
    scrollToAlignment="auto"
    scrollToIndex={-1}
    style={Object {}}
    width={100}
  >
    <Grid
      aria-label="grid"
      aria-readonly={true}
      autoContainerWidth={true}
      autoHeight={false}
      autoWidth={false}
      cellRangeRenderer={[Function]}
      cellRenderer={[Function]}
      className="ReactVirtualized__List"
      columnCount={1}
      columnWidth={100}
      containerRole="rowgroup"
      containerStyle={Object {}}
      estimatedColumnSize={100}
      estimatedRowSize={30}
      getScrollbarSize={[Function]}
      height={100}
      noContentRenderer={[Function]}
      noRowsRenderer={[Function]}
      onRowsRendered={[Function]}
      onScroll={[Function]}
      onScrollbarPresenceChange={[Function]}
      onSectionRendered={[Function]}
      overscanColumnCount={0}
      overscanIndicesGetter={[Function]}
      overscanRowCount={10}
      role="grid"
      rowCount={3}
      rowHeight={20}
      rowRenderer={[Function]}
      scrollToAlignment="auto"
      scrollToColumn={-1}
      scrollToIndex={-1}
      scrollToRow={-1}
      scrollingResetTimeInterval={150}
      style={Object {}}
      tabIndex={0}
      width={100}
    >
      <div
        aria-label="grid"
        aria-readonly={true}
        className="ReactVirtualized__Grid ReactVirtualized__List"
        onScroll={[Function]}
        role="grid"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "boxSizing": "border-box",
            "direction": "ltr",
            "height": 100,
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "relative",
            "width": 100,
            "willChange": "transform",
          }
        }
        tabIndex={0}
      >
        <div
          className="ReactVirtualized__Grid__innerScrollContainer"
          role="rowgroup"
          style={
            Object {
              "height": 60,
              "maxHeight": 60,
              "maxWidth": 100,
              "overflow": "hidden",
              "pointerEvents": "",
              "position": "relative",
              "width": "auto",
            }
          }
        >
          <HitComponent
            hit={
              Object {
                "name": "foo",
              }
            }
            key="0-0"
            style={
              Object {
                "height": 20,
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <span>
              foo
            </span>
          </HitComponent>
          <HitComponent
            hit={
              Object {
                "name": "bar",
              }
            }
            key="1-0"
            style={
              Object {
                "height": 20,
                "left": 0,
                "position": "absolute",
                "top": 20,
                "width": "100%",
              }
            }
          >
            <span>
              bar
            </span>
          </HitComponent>
          <HitComponent
            hit={
              Object {
                "name": "baz",
              }
            }
            key="2-0"
            style={
              Object {
                "height": 20,
                "left": 0,
                "position": "absolute",
                "top": 40,
                "width": "100%",
              }
            }
          >
            <span>
              baz
            </span>
          </HitComponent>
        </div>
      </div>
    </Grid>
  </List>
</Component>
`;
